tools/enrolment_preprocessor.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+import csv
+import sys
+from argparse import ArgumentParser
+from dataclasses import dataclass
+
+from openpyxl import load_workbook
+from openpyxl.cell import Cell
+
+
+@dataclass
+class User:
+    title: str
+    last_name: str
+    first_name: str
+    email: str
+
+
+def fix_users(users: dict[str, User], title: Cell | None, last_name: Cell, first_name: Cell, email: Cell):
+    imported = User(last_name=last_name.value, first_name=first_name.value, email=email.value, title=title.value or "" if title else "")  # type: ignore[arg-type]  # if schema is correct, all values are strings
+    existing = users.setdefault(imported.email, imported)
+    if existing != imported:
+        print("There is a conflict in the user data.")
+        print(f"existing: {existing}.")
+        print(f"imported: {imported}.")
+        if input("Do you want to keep the existing user? (y/n)")[0].lower() == "n":
+            return
```

Comment:
I think the decisions would have to be based on each individual attribute (title, first name, last name).